Feat/participants tabs

dantgw opened the pull request:
- Create Participants Tab
- Set Winner UI
- Progress Bar for Prize Pool
<img width="863" alt="Screenshot 2024-10-15 at 13 39 56" src="https://github.com/user-attachments/assets/bb27c058-b8cc-4b4f-86b7-f633395b0cfc">

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
🚀 **Deployment Started!** 🚧
| Details | Link |
| --- | --- |
| **Follow Progress** | [Workflow Run](https://github.com/Pool-Escrow/PWA/actions/runs/11339978076) |

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
✅ **Deployment Complete!** 🌐
| Details | Link |
| --- | --- |
| **View Deployment** | [Visit Site](https://pwa-h8rneiju6-pool-team.vercel.app) |

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
✅ **Deployment Complete!** 🌐
| Details | Link |
| --- | --- |
| **View Deployment** | [Visit Site](https://pwa-3pkxomwpc-pool-team.vercel.app) |

ottodevs commented on src/app/(pages)/pool/[pool-id]/_components/pool-details.tsx:
```diff
@@ -39,8 +40,8 @@ export default async function PoolDetails({ pool }: { pool: PoolDetailsDTO }) {
                             data-testid='pool-details-progress'
-                            current={pool.poolBalance}
-                            goal={pool.goal}
+                            current={Number(formatUnits(BigInt(pool.poolBalance), pool.tokenDecimals))}
```
Why do we need to convert to Bigint, then parsing, then converting again to number? This specific part was working perfectly and it is completely unrelated to the participants tabs, can you elaborate on this change?

dantgw replied:
I was using the same component and noticed there was the token decimal parsing issue. This is what I'm seeing on the main branch.
<img width="545" alt="Screenshot 2024-10-24 at 13 19 02" src="https://github.com/user-attachments/assets/cfd186e7-183e-4c10-a680-4b8d25c10058">

ottodevs replied:
Good catch, Jonathan and I also identified it and I am working on a fix. Thanks for pointing out!

ottodevs reviewed (changes requested):
Hey Dan,

Thanks for your work on this! I’ve gone through the PR and wanted to share some feedback to help align it with our goals:

1. **Database Schema Update**  
   It looks like we’re missing the database schema update for the new changes. Could you please ensure it’s included?

2. **Unrelated Changes**  
   There are quite a few unrelated changes bundled in this PR, which makes it hard to focus on the core areas for review. Let’s aim to keep PRs more focused on specific tasks to streamline the process. Could you help explaining the unrelated changes or separating them out?

3. **Errors on Participants List (Regular User)**  
   When navigating to the participants list as a regular user, I’m seeing the following errors:

   ![Error Screenshot](https://github.com/user-attachments/assets/46e2cb65-ff1c-451c-9640-e574f9b559c2)

   Let’s make sure to resolve these and check the logs for any similar issues that may arise.

4. **UI Adjustments (Participant View)**  
   - The title "Manage Participants" should not be visible to regular participants.
   - If there’s only one tab, we don’t need to show a tab interface for participants.

5. **Design Alignment with Figma**  
   There are some discrepancies between the current implementation and the Figma designs, such as the search bar, tab bar, and address formatting. It would be great to address these for consistency.

   ![Design Comparison](https://github.com/user-attachments/assets/14459f48-a85a-449e-81b2-6bfab135050e)  
   For example, the progress bar should have a blue background, as seen in the Figma designs:

   ![Progress Bar Example](https://github.com/user-attachments/assets/44b0b000-dee6-405f-b74e-556133c909c5)

6. **Admin Payout Screen Delay**  
   When accessing the payout screen as an admin, there’s a noticeable delay after first seeing a "not authorized" screen, then it resolves after 1-2 seconds. This is inconsistent with other actions like the "create pool" button, which is instant. Could we investigate why the admin status check is taking longer here?

7. **Payout Screen Design Differences**  
   Similarly, the payout screen has several differences compared to the Figma designs. It would be great to bring these in line with the mockups for a more cohesive experience.

---

Thanks again! let me know if you want to discuss any item.

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
✅ **Deployment Complete!** 🌐
| Details | Link |
| --- | --- |
| **View Deployment** | [Visit Site](https://pwa-q05xjklbf-pool-team.vercel.app) |

dantgw commented on next.config.mjs:
```diff
@@ -80,6 +80,7 @@ export default withBundleAnalyzer(
                 {
                     source: '/(.*)',
                     headers: [
+                        // TODO: Enable CSP for production.
```
When I started working on the branch, I was facing CSP issues so I temporarily disabled it. I left the comment there to remind myself to enable in the future. I'll remove the comment since the CSP has been fixed.

dantgw commented on src/app/(pages)/pool/[pool-id]/_components/pool-details-progress.tsx:
```diff
@@ -4,9 +4,14 @@ import { Progress } from '@/app/_components/ui/progress'
 interface PoolDetailsProgressProps {
     current: number
     goal: number
+    description?: string
```
I was using the same progress bar component as the pool details one which had a different text. Seeing as we want to change the color as well, I've created a separate component.

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
✅ **Deployment Complete!** 🌐
| Details | Link |
| --- | --- |
| **View Deployment** | [Visit Site](https://pwa-k3p4shdco-pool-team.vercel.app) |

dantgw commented on src/types/db.ts:
```diff
@@ -1,6 +1,31 @@
 export type Json = string | number | boolean | null | { [key: string]: Json | undefined } | Json[]
 
 export type Database = {
+    graphql_public: {
```
Ok, I updated it. I generated it from the dashboard.

dantgw commented:
I addressed most of the issues in the latest push. I think left with the payout participant page which is not exactly as the figma file. I haven't added the text field as a reason for winning because we have to agree on the changes to the DB.
Some of the UI parts are not so straightforward. 
- Like the button appearing directly above the keyboard
- The curved edges of the selected tab underline which coincides with the line.

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
✅ **Deployment Complete!** 🌐
| Details | Link |
| --- | --- |
| **View Deployment** | [Visit Site](https://pwa-1ox9zgpxk-pool-team.vercel.app) |

ottodevs commented:
<!-- deploying-comment: do not delete/edit this line -->
✅ **Deployment Complete!** 🌐
| Details | Link |
| --- | --- |
| **View Deployment** | [Visit Site](https://pwa-e0ioqttqb-pool-team.vercel.app) |

ottodevs reviewed (approved):
LGTM.
